### PR TITLE
Tweak tests and update README for lodash 3.0 - closes #15

### LINKED ISSUE #15: Review for lodash v3
Lo-Dash v3 has several changes in support for custom builds.

* It allows `lodash modularize exports=es6`
* It drops support for `lodash underscore`
* It has singular category names like `array` instead of `arrays`<br>(though I'm adding aliases for the old style)
* Some methods have changed category

grunt-lodash should be reviewed for compatibility.

### PR DESCRIPTION
**Pull request by phated**
# Don't Merge

This PR needs to be updated once lodash-cli@3.0 is published.  The travis.yml will need to be updated to test that version.

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -119,6 +94,68 @@ config.watch = {
...
+  config.minifyShortFlags = [
+    '-m',
+    'm'
+  ];
+  config.sourceMapShortFlags = [
+    '-p',
+    'p'
+  ];
+  config.modifiers.push(
+    'underscore',
```
Comment (jdalton) [inline config.js]: alphabetic order?

File: test/modularize_test.js [hunk 2]
```diff
@@ -40,38 +40,51 @@ var ignoreFiles = [
...
+    'collection.js',
+    'date.js',
+    'function.js',
+    'lang.js',
+    'number.js',
+    'object.js',
+    'string.js',
+    'support.js',
+    'utility.js'
+  ]
```
Comment (jdalton) [inline test/modularize_test.js]: Nice!

File: config.js (hunk 1 above)
Comment (phated) [inline config.js]: updated

File: test/modularize_test.js (hunk 2 above)
Comment (jdalton) [inline test/modularize_test.js]: Could it be simplified as the folder structure mimics the filename.

Comment (phated) [inline test/modularize_test.js]: probably but if it ever changed, that would be a pain to deal with

File: test/modularize_test.js [hunk 3]
```diff
@@ -84,6 +97,20 @@ var npmModularizeFiles = {
     '_maxpoolsize', '_objectpool', '_objecttypes', '_reescapedhtml', '_reinterpolate',
     '_releasearray', '_releaseobject', '_reunescapedhtml', '_setbinddata',
     '_shimisplainobject', '_shimkeys', '_slice', '_unescapehtmlchar'
+  ],
+  '3.0.0': [
+    '_arraycopy', '_cacheindexof', '_arrayeach', '_charsleftindex', '_arrayevery',
```
Comment (jdalton) [inline test/modularize_test.js]: alphabetic order?

**Comment by jdalton** [comment]
Rock reviewed!

File: test/modularize_test.js (hunk 3 above)
Comment (phated) [inline test/modularize_test.js]: ugh, already a pain retrieving these :( maybe I can bring them into node and sort

Comment (phated) [inline test/modularize_test.js]: worked

Comment (jdalton) [inline test/modularize_test.js]: yap, I do that in the dev console in the browser a lot ;P

Comment (jdalton) [inline test/modularize_test.js]: when they're in order it's easier to spot issues.

**Pull request closed without merging**